converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/FhirTransactionPipeline.cs

```diff
@@ -23,100 +33,161 @@ public class FhirTransactionPipeline : IFhirTransactionPipeline
+                                bundle.Entry.Add(CreateRequestBundleEntryComponent(requestEntry));
+                                useCount++;
+                            }
+
+                            usedPropertyAccessors.Add((propertyAccessor, useCount));
+                        }
+
+                        if (bundle.Entry.Count == 0)
+                        {
+                            // Nothing to update.
+                            return;
+                        }
+
+                        // Execute the transaction.
+                        Bundle responseBundle = await _fhirTransactionExecutor.ExecuteTransactionAsync(bundle, cancellationToken);
+
+                        // Process the response.
+                        int processedResponseItems = 0;
+
+                        foreach ((FhirTransactionRequestResponsePropertyAccessor accessor, int count) in
+                                 usedPropertyAccessors.Where(x => x.Count > 0))
+                        {
+                            var responseEntries = new List<FhirTransactionResponseEntry>();
+                            for (int j = 0; j < count; j++)
+                            {
+                                FhirTransactionResponseEntry responseEntry = CreateResponseEntry(responseBundle.Entry[processedResponseItems + j]);
+                                responseEntries.Add(responseEntry);
+                            }
+
+                            processedResponseItems += count;
+                            accessor.ResponseEntrySetter(context.Response, responseEntries);
+                        }
+
+                        // Execute any additional checks of the response.
+                        foreach (IFhirTransactionPipelineStep pipeline in _fhirTransactionPipelines)
+                        {
+                            pipeline.ProcessResponse(context);
+                        }
+
+                        _logger.LogInformation("Successfully processed the change feed entry.");
+                    }
+                    catch (TaskCanceledException ex) when (!cancellationToken.IsCancellationRequested)
+                    {
+                        throw new RetryableException(ex);
+                    }
+                    catch (HttpRequestException ex)
+                    {
+                        throw new RetryableException(ex);
+                    }
+                },
+                context,
+                cancellationToken);
         }
-
-        // Execute the transaction.
-        Bundle responseBundle = await _fhirTransactionExecutor.ExecuteTransactionAsync(bundle, cancellationToken);
-
-        // Process the response.
-        int processedResponseItems = 0;
-
-        foreach ((FhirTransactionRequestResponsePropertyAccessor accessor, int count) in
-            usedPropertyAccessors.Where(x => x.Count > 0))
+        catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
```

Comment:
I think Polly will handle this gracefully without the need for our own try/catch